### PR TITLE
Fixes after audit kickoff

### PR DESCRIPTION
**Pull request by dorin-iancu**
Add "removeTokenFromWhitelist" functionality in MultiTransferEsdt SC

Fix "block nonce diff" functionality:
- Before: getNextBatch would return a batch where the first and last tx from the batch would not exceed a certain block_nonce diff
- Now: Next pending transaction is returned only if a certain number of blocks have passed since its submission

### CODE REVIEW

File: esdt-safe/src/lib.rs [hunk 1]
```diff
@@ -8,14 +8,23 @@ elrond_wasm::imports!();
 elrond_wasm::derive_imports!();
 
 const DEFAULT_MAX_TX_BATCH_SIZE: usize = 10;
-const DEFAULT_MAX_BLOCK_NONCE_DIFF: u64 = 100;
+const DEFAULT_MIN_BLOCK_NONCE_DIFF: u64 = 100;
```
Comment (sasurobert) [inline esdt-safe/src/lib.rs:11]: this is way too much. put 5.

File: esdt-safe/src/lib.rs [hunk 2]
```diff
@@ -94,28 +103,22 @@ pub trait EsdtSafe {
     fn get_next_transaction_batch(&self) -> SCResult<EsdtSafeTxBatch<Self::BigUint>> {
         self.require_caller_owner()?;
 
-        if self.pending_transaction_address_nonce_list().is_empty() {
-            return Ok(EsdtSafeTxBatch {
-                batch_id: 0,
-                transactions: Vec::new()
-            })
+        let current_block_nonce = self.blockchain().get_block_nonce();
+        let first_tx_block_nonce = match self.get_first_pending_transaction_block_nonce() {
```
Comment (sasurobert) [inline esdt-safe/src/lib.rs:107]: this seems wrong. What we want is that every transaction which is chosen for the next batch is at least Min_BLOCKS old.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: esdt-safe/src/lib.rs (hunk 2 above)
Comment (dorin-iancu) [reply esdt-safe/src/lib.rs:107]: Transactions in the pending list are in a FIFO order of their creation time, so if the first one is old enough, the rest are guaranteed to be as least as old as that one.

File: esdt-safe/src/lib.rs (hunk 1 above)
Comment (dorin-iancu) [reply esdt-safe/src/lib.rs]: Done.

File: esdt-safe/src/lib.rs (hunk 2 above)
Comment (dorin-iancu) [reply esdt-safe/src/lib.rs:107]: ^False. Will fix.

Comment (dorin-iancu) [reply esdt-safe/src/lib.rs]: Fixed.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**